client/js/shout.js

```diff
@@ -644,10 +644,12 @@ $(function() {
 		var button = sidebar.find(".chan[data-target=" + target + "]");
 		var isQuery = button.hasClass("query");
 		var type = msg.type;
+		var self = msg.self;
 		var highlight = type.contains("highlight");
 		var message = type.contains("message");
+		var whois = type.contains("whois");
 		var settings = $.cookie("settings") || {};
-		if (highlight || isQuery || (settings.notifyAllMessages && message)) {
+		if ((highlight || isQuery || (settings.notifyAllMessages && message)) && !self && !whois) {
```

Comment:
True, I could probably change that part a bit since it's now doing some checks twice (such as whether or not the channel receiving the message is active or not). Once for the notification, and once for the badge.